arch/RISCV/RISCVDisassembler.c

```diff
@@ -0,0 +1,417 @@
+{
+  	if (Imm == 0)
+    		return MCDisassembler_Fail;
+  	return decodeSImmOperand(Inst, Imm, Address, Decoder, N);
+}
+
+///
+static DecodeStatus decodeSImmOperandAndLsl1(MCInst *Inst, uint64_t Imm,
+                                             int64_t Address,
+                                             const void *Decoder,
+					     unsigned N) 
+{
+  	//assert(isUInt<N>(Imm) && "Invalid immediate");
+  	// Sign-extend the number in the bottom N bits of Imm after accounting for
+  	// the fact that the N bit immediate is stored in N-1 bits (the LSB is
+  	// always zero)
+  	//Inst.addOperand(MCOperand::createImm(SignExtend64<N>(Imm << 1)));
+  	MCOperand_CreateImm0(Inst, SignExtend64(Imm << 1, N));
+  	return MCDisassembler_Success;
+}
+
+///
+static DecodeStatus decodeCLUIImmOperand(MCInst *Inst, uint64_t Imm,
+                                         int64_t Address,
+                                         const void *Decoder) 
+{
+  	//assert(isUInt<6>(Imm) && "Invalid immediate");
+  	if (Imm > 31) {
+    		Imm = (SignExtend64(Imm, 6) & 0xfffff);
+  	}
+  	//Inst.addOperand(MCOperand::createImm(Imm));
+  	MCOperand_CreateImm0(Inst, Imm);
+  	return MCDisassembler_Success;
+}
+
+///
+static DecodeStatus decodeFRMArg(MCInst *Inst, uint64_t Imm,
+                                 int64_t Address,
+                                 const void *Decoder) 
+{
+  	//assert(isUInt<3>(Imm) && "Invalid immediate");
+  	if (!RISCVFPRndMode_isValidRoundingMode(Imm))
+    		return MCDisassembler_Fail;
+
+  	//Inst.addOperand(MCOperand::createImm(Imm));
+  	MCOperand_CreateImm0(Inst, Imm);
+  	return MCDisassembler_Success;
+}
+
+
+#define GET_SUBTARGETINFO_ENUM
+#include "RISCVGenSubtargetInfo.inc"
+#include "RISCVGenDisassemblerTables.inc"
+
+/// 
+static void clear_MI_insn_detail(MCInst *MI) 
+{
+  	if (MI->flat_insn->detail) {
+    		memset(MI->flat_insn->detail, 0, sizeof(cs_detail));
+  	}
+
+  	return;
+}
+
+///
+static DecodeStatus RISCVDisassembler_getInstruction(int mode, MCInst *MI,
+				 const uint8_t *code, size_t code_len,
+				 uint16_t *Size, uint64_t Address,
+				 MCRegisterInfo *MRI) 
+{
+  	// TODO: This will need modification when supporting instruction set
+  	// extensions with instructions > 32-bits (up to 176 bits wide).
+  	uint32_t Inst = 0;
+  	DecodeStatus Result;
+
+  	// It's a 32 bit instruction if bit 0 and 1 are 1.
+  	if ((code[0] & 0x3) == 0x3) {
+      		if (code_len < 4) {
+        		*Size = 0;
+        		return MCDisassembler_Fail;
+      		}
+
+      	*Size = 4;
+      	// Get the four bytes of the instruction.
+      	//Encoded as little endian 32 bits.
+      	Inst = code[0] | (code[1] << 8) | (code[2] << 16) | (code[3] << 24);
+      	clear_MI_insn_detail(MI);
+      	Result = decodeInstruction(DecoderTable32, MI, Inst, Address, MRI, mode);
+  	} else {
+    		if (code_len < 2) {
+      			*Size = 0;
+      			return MCDisassembler_Fail;
+    		}
+
+    		if (!getFeatureBits(mode)) {
```

Comment:
Because of [getFeatureBits](https://github.com/aquynh/capstone/pull/1401/files#diff-e73e415b1ee781afd1d70b29f9ac9885R35), this condition is always false.